### DISCUSSION TITLE
If initialItemCount is specified, rendering will be very slow.

### DISCUSSION THREAD

**Discussion by unpori** [petyosi/react-virtuoso#708] [Q&A]
I am using SSR in Next.js and am trying to use initialItemCount to support SSR, but
For some reason, specifying initialItemCount slows down the initial rendering.

If I remove initialItemCount, the rendering speed is a little faster.
What could be the cause of these problems?

Also, I need to use this with VirtuosoGrid, but
defaultItemHeight does not seem to be supported by VirtuosoGrid yet.

How do I get VirtuosoGrid to support SSR?

**Comment by vsMdulcio** [comment]
Getting this same thing, but my usecase is if I don't specify the initialItemCount, my items don't get rendered (in a popout window)

@petyosi
